recipes/linux-headers-generic/all/conanfile.py

```diff
@@ -0,0 +1,40 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class LinuxHeadersGenericConan(ConanFile):
+    name = "linux-headers-generic"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.kernel.org/"
+    license = "GPL-2.0-only"
+    description = "Generic Linux kernel headers"
+    topics = ("linux", "headers", "generic")
+    settings = "os", "arch"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("linux-headers-generic supports only Linux")
+        if tools.cross_building(self):
+            raise ConanInvalidConfiguration("linux-headers-generic can not be cross-compiled")
+
+    def build_requirements(self):
+        self.build_requires("make/4.3")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def build(self):
+        with tools.chdir(os.path.join(self._source_subfolder)):
+            self.run("{} headers".format(tools.get_env("CONAN_MAKE_PROGRAM")))
```

Comment:
This will also automatically enable parallel building, if possible.
```suggestion
            autotools = AutotoolsBuildEnvironment(self)
            autotools.make(target="headers")
```